sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1192,11 +1192,23 @@ class Analyzer(
             if findAliases(projectList).intersect(conflictingAttributes).nonEmpty =>
           Seq((oldVersion, oldVersion.copy(projectList = newAliases(projectList))))
 
+        // We don't need to search child plan recursively if the projectList of a Project
+        // is only composed of Alias and doesn't contain any conflicting attributes.
+        // Because, even if the child plan has some conflicting attributes, the attributes
+        // will be aliased to non-conflicting attributes by the Project at the end.
+        case _ @ Project(projectList, _)
+          if findAliases(projectList).size == projectList.size =>
+          Nil
```

Comment:
Don't we need to put this before previous `Project` pattern?

Reply:
I think we don't. It's mutually exclusive with the previous one as it has an implicit requirement ( `findAliases(projectList).intersect(conflictingAttributes).isEmpty`). Therefore, the actual requirement here is:
`findAliases(projectList).intersect(conflictingAttributes).isEmpty && findAliases(projectList).size == projectList.size`